src/renderer/components/+catalog/details-for-catalog-entity.injectable.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import type { CatalogEntity } from "../../../common/catalog";
+import detailItemRegistrationsInjectable from "./detail-item-registrations.injectable";
+import { conforms, eq, includes } from "lodash/fp";
+
+const detailsForCatalogEntityInjectable = getInjectable({
+  instantiate: (di, catalogEntity: CatalogEntity) => {
+    const allRegistrations = di.inject(detailItemRegistrationsInjectable);
+
+    const registrationsForEntity = allRegistrations.get().filter(
+      conforms({
+        kind: eq(catalogEntity.kind),
+        apiVersions: includes(catalogEntity.apiVersion),
+      }),
+    );
+
+    return {
+      defaultIsShown: !registrationsForEntity.some(
+        (registration) => registration.priority > 999,
+      ),
+
+      components: registrationsForEntity.map(
+        (registration) => registration.components.Details,
+      ),
+    };
+  },
+
+  lifecycle: lifecycleEnum.transient,
```

Comment:
Random question: if i replace lifecycle to `lifecycleEnum.singleton` how it would it break this injectable usages?

Reply:
Two things:

1. Injectable would throw because we would be using `singleton` with `instantiationParameter`. (see below)
2. (if 1. defensive code would not be present) CatalogEntity used would be same for every time when injected.

However, this is something that we are actually working on, because we see this bad design. 

This should be `singleton` but only in scope of `instantiationParameter`. So if you give `"some-string"` and `"some-other-string"` as `instantiationParameter`, you will get two different instances, but if you inject `"some-string"` twice, you will get same instance. This allows us to get rid of these type of transient lifecycle usages and start using `computed` as a mechanism for achieving reactive UI.